e2e/cypress/integration/interactive_button/interactive_button_spec.js

```diff
@@ -0,0 +1,106 @@
+            };
+
+            cy.apiCreateWebhook(newIncomingHook).then((hook) => {
+                incomingWebhook = hook;
+            });
+        });
+    });
+
+    it('IM10164 - Include extra metadata when clicking an interactive button', () => {
+        // # Post an incoming webhook
+        cy.postIncomingWebhook({url: incomingWebhook.url, data: payload});
+
+        // # Get message attachment from the last post
+        let interactiveButtonPostId;
+        cy.getLastPostId().then((postId) => {
+            interactiveButtonPostId = postId;
+            cy.get(`#messageAttachmentList_${postId}`).as('messageAttachmentList');
+        });
+
+        // * Verify each element of message attachment list and then click "Ephemeral" button
+        cy.get('@messageAttachmentList').within(() => {
+            cy.get('.attachment__thumb-pretext').should('be.visible').and('have.text', 'This is the attachment pretext.');
+            cy.get('.post-message__text-container').should('be.visible').and('have.text', 'This is the attachment text for Interactive Message Buttons.');
+            cy.findByText('Ephemeral Message').should('be.visible').click();
+        }).then(() => {
+            // * Verify metadata on click to "Ephemeral" button
+            verifyMessage('Ephemeral', 'do_something_ephemeral', interactiveButtonPostId);
+        });
+
+        // * Verify "Update" button on message attachment list and then click
+        cy.get('@messageAttachmentList').within(() => {
+            cy.findByText('Update').should('be.visible').click();
+        }).then(() => {
+            // * Verify metadata on click to "Update" button
+            verifyMessage('Update', 'do_something_update', interactiveButtonPostId);
+        });
+    });
+});
+
+function verifyMessage(action, context, interactiveButtonPostId) {
+    cy.getLastPostId().then((postId) => {
+        cy.get(`#postMessageText_${postId}`).should('be.visible').within(() => {
+            cy.findByText(`${action} | context:`).should('be.visible');
+            cy.getCurrentUserId().then((userId) => {
+                cy.findByText(`user_id: "${userId}"`).should('be.visible');
+                cy.findByText('user_name: "sysadmin"').should('be.visible');
+            });
+            cy.findByText('channel_name: "town-square"').should('be.visible');
+            cy.apiGetChannelByName('ad-1', 'town-square').then((channel) => {
+                cy.findByText(`channel_id: "${channel.id}"`).should('be.visible');
+                cy.findByText('channel_name: "town-square"').should('be.visible');
+                cy.findByText(`team_id: "${channel.team_id}"`).should('be.visible');
+                cy.findByText('team_domain: "ad-1"').should('be.visible');
+            });
+            cy.findByText(`post_id: "${interactiveButtonPostId}"`).should('be.visible');
+            cy.findByText(/trigger_id:/).should('be.visible');
+            cy.findByText('type: ""').should('be.visible');
```

Comment:
Hm, I'm surprised by this. The expected behaviour from the top of my head should be that `button` is returned, even if it's not set.

If you agree, would you open a bug ticket to fix this on the server?

Reply:
Sure, will check that.